src/domain.rs

```diff
@@ -0,0 +1,31 @@
+/// Configuration for domain checks to be performed on either the `Host`
+/// or the `Origin` header.
+#[derive(Debug)]
+pub enum DomainCheck<Domain = String> {
+	/// Allow any domain
+    AllowAny,
+    /// Allow a domain if it is one on the list
+    AllowList(Vec<Domain>),
+    /// Allow a domain if it is NOT on the list
+    DenyList(Vec<Domain>),
+}
+
+impl<Domain> Default for DomainCheck<Domain> {
+    fn default() -> Self {
+        DomainCheck::AllowAny
+    }
+}
+
+impl<Domain> DomainCheck<Domain>
+where
+	Domain: AsRef<str>,
+{
+	/// Checks if a `domain` is allowed the handshake
+    pub(crate) fn is_allowed(&self, domain: &[u8]) -> bool {
+        match self {
+            DomainCheck::AllowAny => true,
+            DomainCheck::AllowList(list) => list.iter().any(|d| d.as_ref().as_bytes() == domain),
```

Comment:
If a host is on both the allow and deny list it will be allowed. Maybe not a problem but worth pointing out in the docs?